aten/src/ATen/native/xnnpack/Common.h

```diff
@@ -9,6 +9,47 @@
 namespace at {
 namespace native {
 namespace xnnpack {
+
+struct Deleter final {
+  void operator()(const xnn_operator_t op) const {
+    xnn_delete_operator(op);
+  }
+};
+
+using Operator = std::unique_ptr<xnn_operator, Deleter>;
+
+struct Context final {
```

Comment:
So the reason to expose this to Common.h is because the context has to be stored in the new custom class, e.g. LinearOpContext. It might be possible to move it back where it was but that would require custom class registration to be moved along with it, which may make code organization little unclean. 
Regarding, having separate Context per op: The reason for me to do it this was mainly because in XNNPACK there is a single operator context. I am open to both options, via separate op specific context as well as union approach, but I prefer to keep that in Common.h for the reasons mentioned above. Union approach might be preferable to be sort of in sync with what XNNPACK does?